packages/block-editor/src/components/link-control/is-url-image.js

```diff
@@ -0,0 +1,9 @@
+/**
+ * Checks if the url is an image.
+ * @param {string} url the url we are checking.
+ * @return {boolean} true if the url is an image url.
+ */
+export default function isImageUrl( url ) {
+	const pattern = /(http(s?):)([/|.|\w|\s|-])*\.(?:jpg|gif|png)/g;
+	return pattern.test( url );
+}
```

Comment:
Some simple tests for this would be great.

Reply:
I just saw that there is another PR (https://github.com/WordPress/gutenberg/pull/57824) open for the same thing and I prefer how they implement this check using mimetypes instead of a regex. I was going to grab that implementation myself

Reply:
I spoke too soon! seems like it can be done for the suggestions because they come from `fetchLinkSuggestions` but the link previews don't have that, so we'd still have to figure out if the URL is an image via proding at the actual URL. I'll go about doing some tests for this function then.